### ISSUE TITLE
Assertion Failure 

### ISSUE THREAD

**Issue by ashermancinelli** [LLNL/hiop#171]
When developing ExaGO we moved to the new hiop interface currently in develop branch. After what appeared to be a successful refactor, the build failed only on intel system (marianas) with hiop develop branch (commit hashes `2e927e4` or `dbb4a7` for example) built in debug mode. Builds passed on power 9 systems (newell and ascent). Pipeline #4584 is an example [(linked here).](https://gitlab.pnnl.gov/exasgd/frameworks/exago/-/pipelines/4584) This failure went away when we rebuilt hiop in release mode.

Assertion:
```console
47: opflow: /people/manc568/projects/hiop/src/Optimization/hiopPDPerturbation.hpp:184: 
    bool hiop::hiopPDPerturbation::compute_perturb_singularity(double&, double&, double&, double&):
    Assertion `delta_cc == 0. && delta_cd == 0.' failed.
```

CC @cnpetra @pelesh @abhyshr

**Comment by nychiang** [comment]
@ashermancinelli To reproduce this error, I am trying to install HiOP and ExaGO on ascent.
Do you have a project-specific module list for ascent, like what we have for summit? 
I think I found the bug, but I would like to test my fix on ascent before pushing the fix to the repository.

**Comment by pelesh** [comment]
@cnpetra, is the assertions supposed to be
```c++
assert(delta_cc == 0. && delta_cd == 0.);
```
or
```c++
assert(delta_cc_ == 0. && delta_cd_ == 0.);
```
?

Other assert statements in function `compute_perturb_singularity` check member variables `delta_cc_` and `delta_cd_`, while the offending assertion checks input arguments of the function `delta_cc` and `delta_cd`.

**Comment by cnpetra** [comment]
> @cnpetra, is the assertions supposed to be
> 
> ```c++
> assert(delta_cc == 0. && delta_cd == 0.);
> ```
> 
> or
> 
> ```c++
> assert(delta_cc_ == 0. && delta_cd_ == 0.);
> ```
> 
> ?
> 
> Other assert statements in function `compute_perturb_singularity` check member variables `delta_cc_` and `delta_cd_`, while the offending assertion checks input arguments of the function `delta_cc` and `delta_cd`.

indeed, that looks like the problem, see @nychiang comments...

**Comment by cnpetra** [comment]
closed by PR#174
